File: app/src/main/java/closer/vlllage/com/closer/store/models/Event.java
package closer.vlllage.com.closer.store.models;

import java.util.Date;

import io.objectbox.annotation.Entity;

@Entity
public class Event extends BaseObject {
    private String name;
    private String about;
    private String groupId;
    private Double latitude;
    private Double longitude;
    private Date startsAt;
    private Date endsAt;
    private boolean cancelled;

    public String getName() {
        return name;
    }

    public Event setName(String name) {
        this.name = name;
        return this;
    }

    public Double getLatitude() {
        return latitude;
    }

    public Event setLatitude(Double latitude) {
        this.latitude = latitude;
        return this;
    }

    public Double getLongitude() {
        return longitude;
    }

    public Event setLongitude(Double longitude) {
        this.longitude = longitude;
        return this;
    }

    public String getAbout() {
        return about;
    }

    public Event setAbout(String about) {
        this.about = about;
        return this;
    }

    public Date getStartsAt() {
        return startsAt;
    }

    public Event setStartsAt(Date startsAt) {
        this.startsAt = startsAt;
        return this;
    }

    public Date getEndsAt() {
        return endsAt;
    }

    public Event setEndsAt(Date endsAt) {
        this.endsAt = endsAt;
        return this;
    }

    public boolean isCancelled() {
        return cancelled;
    }

    public Event setCancelled(boolean cancelled) {
        this.cancelled = cancelled;
        return this;
    }

    public String getGroupId() {
        return groupId;
    }

    public Event setGroupId(String groupId) {
        this.groupId = groupId;
        return this;
    }

    public boolean hasGroup() {
        return groupId != null;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/store/models/Event.kt
package closer.vlllage.com.closer.store.models

import closer.vlllage.com.closer.handler.event.EventReminder
import closer.vlllage.com.closer.store.EventRemindersListJsonConverter
import io.objectbox.annotation.Convert
import io.objectbox.annotation.Entity
import java.util.*

@Entity
class Event : BaseObject() {
    var name: String? = null
    var about: String? = null
    var isPublic: Boolean = false
    var groupId: String? = null
    var creator: String? = null
    var latitude: Double? = null
    var longitude: Double? = null
    var startsAt: Date? = null
    var endsAt: Date? = null
    var allDay: Boolean = false
    var cancelled: Boolean = false

    @Convert(converter = EventRemindersListJsonConverter::class, dbType = String::class)
    var reminders: List<EventReminder>? = null
}
